torch_xla/csrc/ops/ops.cpp

```diff
@@ -415,6 +416,42 @@ NodePtr BroadcastTensors(tensorflow::gtl::ArraySlice<const Value> tensors) {
                    std::move(lower_fn), /*num_outputs=*/tensors.size());
 }
 
+NodePtr Norm(const Value& input, c10::optional<at::Scalar> p,
+             c10::optional<at::ScalarType> dtype, at::IntArrayRef dim,
+             bool keepdim) {
+  std::vector<xla::int64> dimensions(dim.begin(), dim.end());
+  if (dimensions.empty()) {
+    dimensions = xla::util::Iota<xla::int64>(input.shape().rank());
+  }
+  if (!p.has_value() || p->toDouble() == 2.0) {
+    NodePtr square = input * input;
+    NodePtr result = MakeNode<Sum>(square, dimensions, keepdim, dtype);
+    return Sqrt(result);
+  }
+  double norm_value = p->toDouble();
+  if (norm_value == 1.0) {
+    // Contrary to documentation, norm(p=1) has nothing to do with traces and
```

Comment:
Isn't the nuclear norm the sum of singular values? I'm equally confused by the documentation, I guess we'll go with this.

Reply:
> Isn't the nuclear norm the sum of singular values? I'm equally confused by the documentation, I guess we'll go with this.

Oh, looking *at their code* it is clearer!
We need to trace(SVD(x)) it.
Doing it ...